### ISSUE TITLE
Player Position

### ISSUE THREAD

**Issue by machine7331** [MrEliasen/Supremacy-Framework#11]
@MrEliasen I'm trying to add player positioning saving to the database also. However, the data is returned as the position but in the database it is 0. Could you help walk me through this a bit? Maybe skype?

**Comment by MrEliasen** [comment]
I'll try guide you the best I can.

You need to modify this file: https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy.altis/compile/player/syncPlayerData.sqf - update the "stats" and "everything" array.
It is important that you append the player position to the arrays, so the rest of the array remains in the same order. Failing to do so will result in player saving not working correctly.

Then edit this file: https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy_server/3rdparty/extDB2/dbCall.sqf, updating the "saveplayerstats" and "saveplayerfull" to add the player position you appended to the data array to the SQL query.

Now, in the https://github.com/MrEliasen/Supremacy-Framework/blob/master/Dist/%40extDB2/extDB/sql_custom_v2/spmc.ini, update the SQL queries, remember to append the data to the insert and update queries where applicable.
(remember to fetch the position in the "loadPlayer" query as well).

Update the https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy_server/compile/player/svrRequestPlayerData.sqf file, added the fetched player location from the query (Add it to the array in on the bottom line).

In https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy.altis/compile/player/processPlayerData.sqf, set the players new position (using [_playerPosition] spawn SPMC_fnc_spawnPlayer;).

In https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy.altis/compile/player/getPlayerSpawn.sqf
Remove the Switch statement.

This is how I believe it will be done, however, I have not tried but let me know if it helps. I am working at the moment, so if I do not reply just give it a few hours.

**Comment by machine7331** [comment]
Thank you, I have done everything but this part:
"Update the https://github.com/MrEliasen/Supremacy-Framework/blob/master/Source/supremacy_server/compile/player/svrRequestPlayerData.sqf file, added the fetched player location from the query (Add it to the array in on the bottom line)."

This is where i think i had messed up. I'll give it a shot

**Comment by MrEliasen** [comment]
No worries, let me know if you are still having issues and I will look into it when I am back home.
If you want to check if the desired data is available, use [diag_log](https://community.bistudio.com/wiki/diag_log) eg. `diag_log _pos;` and it will show up in your server or client logs (depending on where you called it). You can do this with pretty much everyone.

Good luck!

**Comment by MrEliasen** [comment]
Did you managed to get it sorted?

**Comment by machine7331** [comment]
No, the sql query looks correct but i broke it somewhere else now bc the database saves the players name but everything else is just brackets []. I tried appending to the array with pushback and also just inserting it at the end. but both failed. I think i have screwed up around here "Failing to do so will result in player saving not working correctly."

**Comment by MrEliasen** [comment]
Try `diag_log` the sql data before the insert query, it should help you understand what it is trying to do.
I highly recommend using diag_log as much as possible, following the data from end to end to make sure it is as you expect.

You can for example add this `diag_log "============ DEBUG ============"` before you diag_log the data to help you find it in the logs.

Good luck!

**Comment by machine7331** [comment]
finally had time to get back at this and i have it working and saving to the database. Now i have an issue with the player spawn menu still coming up even though the data from the database has been loaded. I'll let you know when i tackle this part now.

**Comment by MrEliasen** [comment]
Find the part of the code which triggers the spawn menu, (you can search for the filename, and you should find it. Should be called something like SPMC_fc_<filename>.

You might need to "hack" this slightly as the player position is meant to only be set based on the spawn type/spawn menu selection. But you could find the spawn menu code as well, remove what shows the menu and instead load in the coords from the database.

If you need help, just let me know :)
